plugins/modules/efs.py

```diff
@@ -736,6 +783,8 @@ def main():
             changed = connection.update_file_system(name, throughput_mode, provisioned_throughput_in_mibps) or changed
         changed = connection.converge_file_system(name=name, tags=tags, purge_tags=purge_tags, targets=targets,
                                                   throughput_mode=throughput_mode, provisioned_throughput_in_mibps=provisioned_throughput_in_mibps) or changed
+        if transition_to_ia:
+            changed = connection.update_lifecycle_policy(name, transition_to_ia)
```

Comment:
```suggestion
            changed |= connection.update_lifecycle_policy(name, transition_to_ia)
```

Using `changed =` will override the changed state from previous updates (potentially back to False).  While it looks like there's a bug in the existing code it's better not to add additional issues that will need fixing while adding a new feature.